esphome/components/remote_base/pronto_protocol.cpp

```diff
@@ -40,6 +40,28 @@ namespace remote_base {
 
 static const char *const TAG = "remote.pronto";
 
+bool ProntoData::operator==(const ProntoData &rhs) const {
+  ESP_LOGD(TAG, "operator==");
+  ESP_LOGD(TAG, "this.data: %s", data.c_str());
+  ESP_LOGD(TAG, " rhs.data: %s", rhs.data.c_str());
+  std::vector<uint16_t> data1 = decode_pronto(data);
+  std::vector<uint16_t> data2 = decode_pronto(rhs.data);
+
+  if (data1.size() != data2.size()) {
+    return false;
+  }
+
+  uint16_t total_diff = 0;
```

Comment:
```suggestion
  uint32_t total_diff = 0;
```

With 16-bit vars this could easily lead to unsigned integer overflow

Reply:
Fixed.